### ISSUE TITLE
xgb.cv gives error, saying it 'found -2147483648 in label.'

### ISSUE THREAD

**Issue by mariekevanbuchem** [dmlc/xgboost#4057]
I'm not sure if this is an XGBoost issue or if I'm doing something wrong, but I keep getting an error from code that used to work. I'm trying to fit an xgboost model on a dataset with 8 columns and 11496 rows (see attached txt file for dataset). 

[Help_on_file.txt](https://github.com/dmlc/xgboost/files/2753203/Help_on_file.txt)

I've created a test and training set, turned them into XGB matrices, and set the parameters. 

```
# XGboost
set.seed(101)
inTrain <- createDataPartition(y = df$V8, p = 0.8, list = F)
training <- df[inTrain,]
testing <- df[-inTrain,]
training$V8 <- as.factor(training$V8)
testing$V8 <- as.factor(testing$V8)

# Make training set and test set
train.data <- training
train.data$V8 <- NULL
train.data <- as.matrix(train.data)
test.data <- testing
test.data$V8 <- NULL
test.data <- as.matrix(test.data)

# Change labels to numeric 
training$V8 <- as.numeric(training$V8)
train.labs <- training$V8 - 1
testing$V8 <- as.numeric(testing$V8)
test.labs <- testing$V8 - 1

train.data.matrix <- as.matrix(train.data)

# Make XGB matrix 
xgb_train <- xgb.DMatrix(data = train.data, label = train.labs)
xgb_val <- xgb.DMatrix(data = test.data, label = test.labs)

# Set parameters(default)
params <- list(booster = "gbtree", objective = "multi:softmax", num_class = 8, eval_metric = "mlogloss")
```

So far, so good. However, when trying to calculate the number of folds for cross validation using xgb.cv, I get the following error:
```
# Calculate # of folds for cross-validation
xgbcv <- xgb.cv(params = params, data = xgb_train, nrounds = 100, nfold = 5, showsd = TRUE, 
stratified = TRUE, print_every_n = 10, early_stop_round = 20, maximize = FALSE, prediction = TRUE)

Error in xgb.iter.update(fd$bst, fd$dtrain, iteration - 1, obj) : 
  [16:03:34] amalgamation/../src/objective/multiclass_obj.cc:78: 
Check failed: label_error >= 0 && label_error < nclass 
SoftmaxMultiClassObj: label must be in [0, num_class), num_class=8 but found -2147483648 in label.

Stack trace returned 10 entries:
[bt] (0) 0   xgboost.so                          0x000000010f5a05f6 dmlc::StackTrace() + 246
[bt] (1) 1   xgboost.so                          0x000000010f5a03bf dmlc::LogMessageFatal::~LogMessageFatal() + 47
[bt] (2) 2   xgboost.so                          0x000000010f5e8aef xgboost::obj::SoftmaxMultiClassObj::GetGradient(xgboost::HostDeviceVector<float>*, xgboost::MetaInfo const&, int, xgboost::HostDeviceVector<xgboost::detail::GradientPairInternal<float> >*) + 2767
[bt] (3) 3   xgboost.so                          0x000000010f5c587e xgboost::LearnerImpl::UpdateOneIter(int, xgboost::DMatrix*) + 894
[bt] (4) 4   xgboost.so                          0x000000010f5bdf47 XGBoosterUpdateOneIter + 87
[bt] (5) 5   xgboost.so              
```

I have no clue where the -2147483648 is coming from...  I've triple checked all the labels, and the minimum is 0 and the maximum is 7, so that should check out. The weird part is that this code used to work, but when trying it again today, it suddenly gave this error. I hope someone can help me!

**Comment by khotilov** [comment]
You have NA's in your label, and -2147483648 is reserved to represent integer NA in R. It might be worth adding a check for NA in labels while constructing a xgb.DMatrix in R.

**Comment by mariekevanbuchem** [comment]
Oh wow thank you, I guess somewhere on the way I removed that check... It works now, thanks for the help! :)
